internal/vault/config.go

```diff
@@ -44,16 +44,27 @@ func MakeVaultClient(ctx context.Context, cfg *ClientConfig, client ctrlclient.C
 
 	var b []byte
 	if cfg.CACertSecretRef != "" {
-		s := &v1.Secret{}
-		if err := client.Get(ctx, types.NamespacedName{
+		objKey := ctrlclient.ObjectKey{
 			Namespace: cfg.K8sNamespace,
 			Name:      cfg.CACertSecretRef,
-		}, s); err != nil {
+		}
+		s := &v1.Secret{}
+		if err := client.Get(ctx, objKey, s); err != nil {
 			return nil, err
 		}
+
 		var ok bool
-		if b, ok = s.Data["ca.crt"]; !ok {
-			return nil, fmt.Errorf(`"ca.crt" was empty in the CA secret %s/%s`, cfg.K8sNamespace, cfg.CACertSecretRef)
+		key := "ca.crt"
```

Comment:
nit: I feel like `ca.crt` has to be a constant somewhere, I couldnt find it though so no biggie!

Reply:
Fixed in https://github.com/hashicorp/vault-secrets-operator/pull/195/commits/c9f007e732bfad9b8003f00f518aad253287a2fa